nodemcu/mispec_ads1115.lua

```diff
@@ -0,0 +1,223 @@
+            mispec.ok(p1 ~= nil, "missing volt")
+            mispec.ok(p2 ~= nil, "missing volt_dec")
+            mispec.ok(p3 ~= nil, "missing raw")
+        end)
+    end)
+
+
+    it:should("cleanup timer when deallocated", function()
+        local id, scl, sda, alert_pin = 0, 1, 2, 3
+        i2c.setup(id, sda, scl, i2c.SLOW)
+        ads1115.reset()
+        local adc2 = ads1115.ads1115(id, ads1115.ADDR_VDD)
+        adc2:setting(ads1115.GAIN_6_144V, ads1115.DR_8SPS, ads1115.DIFF_0_3, ads1115.SINGLE_SHOT)
+        local called = false
+        adc2:startread(function(volt, volt_dec, adc)
+            called = true
+        end)
+        local wait = true
+        mispec.eventually(function()
+            if wait then
+                wait = false
+                mispec.ok(false)
+            end
+        end, 1, 250)
+        mispec.andThen(function()
+            mispec.ko(called == true, "unexpected call")
+        end)
+        adc2 = nil
+    end)
+
+
+    it:should("read in single shot mode with hardware conversion ready", function()
+        local id, scl, sda, alert_pin = 0, 1, 2, 3
+        i2c.setup(id, sda, scl, i2c.SLOW)
+        ads1115.reset()
+        local adc1 = ads1115.ads1015(id, ads1115.ADDR_GND)
+        adc1:setting(ads1115.GAIN_6_144V, ads1115.DR_128SPS, ads1115.DIFF_0_3, ads1115.SINGLE_SHOT, ads1115.CONV_RDY_1)
+        local called = false
+        local volt, volt_dec, adc
+        local function conversion_ready(level, when)
+            gpio.trig(alert_pin)
+            volt, volt_dec, adc = adc1:read()
+            called = true
+        end
+        gpio.mode(alert_pin, gpio.INT)
+        gpio.trig(alert_pin, "down", conversion_ready)
+        adc1:startread()
+        mispec.eventually(function()
+            mispec.ok(called == true, "missing callback")
+            mispec.ok(volt ~= nil, "missing volt")
+            mispec.ok(volt_dec ~= nil, "missing volt_dec")
+            mispec.ok(adc ~= nil, "missing raw")
+        end)
+    end)
+
+
+    it:should("trigger when input value is out of range (window)", function()
+        local id, scl, sda, alert_pin = 0, 1, 2, 3
+        i2c.setup(id, sda, scl, i2c.SLOW)
+        ads1115.reset()
+        local adc1 = ads1115.ads1015(id, ads1115.ADDR_GND)
+        adc1:setting(ads1115.GAIN_1_024V, ads1115.DR_128SPS, ads1115.DIFF_0_3, ads1115.CONTINUOUS, ads1115.COMP_1CONV, -15, 100, ads1115.CMODE_WINDOW)
+        local triggered = false
+        local volt, volt_dec, adc
+        local function comparator_trigger(level, when)
+            volt, volt_dec, adc = adc1:read()
+            gpio.trig(alert_pin)
+            triggered = true
+        end
+        gpio.mode(alert_pin, gpio.INT, gpio.PULLUP)
+        gpio.trig(alert_pin, "down", comparator_trigger)
+    end)
+
+    it:should("trigger when input value is out of range (window) - complete", function()
+        if _G["mcp4725"] == nil then
+            return error("mispec skip: mcp4725 module is missing")
```

Comment:
I think this would be nice as a function of the mispec module.

For example, a `mispec.skip(reason)` which throws an error with that prefix. Then, anyone writing tests wouldn't need to worry about the exact prefix that you need to use so it's considered skipped.
What do you think?